src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java

```diff
@@ -0,0 +1,159 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.commons.util.JsonUtil.readJsonFile;
+import static seedu.address.storage.JsonAdaptedInternship.MISSING_FIELD_MESSAGE_FORMAT;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.item.Internship;
+import seedu.address.model.item.field.Name;
+import seedu.address.model.item.field.Time;
+import seedu.address.testutil.InternshipBuilder;
+import seedu.address.testutil.TypicalInternship;
+
+public class JsonAdaptedInternshipTest {
+    private static final Path TEST_DATA_FOLDER = Paths
+            .get("src", "test", "data", "JsonAdaptedInternshipTest");
+
+    private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
+        return prefsFileInTestDataFolder != null
+                ? TEST_DATA_FOLDER.resolve(prefsFileInTestDataFolder)
+                : null;
+    }
+
+    @Test
+    public void load_jsonToJsonAdaptedInternship_success() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("validInternship.json"),
+                        JsonAdaptedInternship.class);
+        JsonAdaptedInternship expectedJsonAdapatedInternship =
+                new JsonAdaptedInternship(new InternshipBuilder(TypicalInternship.PAYPAL).build());
+        assertEquals(expectedJsonAdapatedInternship, jsonAdaptedInternship.get());
+    }
+
+    @Test
+    public void loadToModel_jsonToInternship_success() throws DataConversionException, IllegalValueException {
```

Comment:
`toModelType()` converts a `JsonAdaptedInternship` to an `Internship`.

I think we should not include reading of json in here.